syntax = "proto3";
import "google/protobuf/empty.proto";

package acs.order_tracking;

enum Status {
    UNKNOWN = 0;
    CREATED = 1;
    IN_PROGRESS = 2;
    COMPLETED = 3;
    CANCELLED = 4;
}

message Destination {
    float x = 1;
    float y = 2;
}

message Order {
    string id = 1;
    string name = 2;
    Status status = 3;
    Destination destination = 4;
}

message CreateOrderRequest {
    string name = 1;
    Destination destination = 2;
}

message CreateOrderResponse {
    Order order = 1;
}

message GetOrderRequest {
    string id = 1;
}

message GetOrderResponse {
    Order order = 1;
}

message UpdateOrderRequest {
    string id = 1;
    Status status = 2;
}

message UpdateOrderResponse {
    Order order = 1;
}

message DeleteOrderRequest {
    string id = 1;
}

message DeleteOrderResponse {
    Order order = 1;
}

service OrderTrackingService {
    rpc CreateOrder(CreateOrderRequest) returns (CreateOrderResponse);
    rpc GetOrder(GetOrderRequest) returns (GetOrderResponse);
    rpc UpdateOrder(UpdateOrderRequest) returns (UpdateOrderResponse);
    rpc DeleteOrder(DeleteOrderRequest) returns (DeleteOrderResponse);
}
